src/sonic-bgpcfgd/bgpcfgd/managers_bgp.py

```diff
@@ -168,9 +167,11 @@ def add_peer(self, vrf, nbr, data):
         bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"]["bgp_asn"]
         #
         lo0_ipv4 = self.get_lo_ipv4("Loopback0|")
-        if lo0_ipv4 is None:
```

Comment:
is that possible the loopback0 interface comes after bgpcfgd start? in that case, if we don't wait for this loopback interface up, it can't get the lo0 ip.

Reply:
It get lo0 ip from config_db https://github.com/sonic-net/sonic-buildimage/blob/3db62a0c6f064cc1e53ecf5f223d5165010dc175/src/sonic-bgpcfgd/bgpcfgd/managers_bgp.py#L324
I don't think there is a scenario that we want loopback0 ipv4 but it is not shown in config_db

Reply:
hi @yaqiangz I mean is there a race condition between bgp startup and the population of loopback config to the config_db? If we remove the dependency, and bgp startup before the loopback config populated into the config_db could cause issue.

Reply:
If I understand correctly, your concern is table subscription for interest table changed? The table change subscription is defined in [main.py](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-bgpcfgd/bgpcfgd/main.py#L49) (DEVICE_METADATA and LOOPBACK_INTERFACE are both subscribed). We didn't change this part. The dependency is a condition for **whether to invoke handler function when subscription table changes (Checked by func `self.directory.available_deps`) [manager.py](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-bgpcfgd/bgpcfgd/manager.py#L41)**. We removed this dependency is to **enable add bgp peer when Loopback v4 address doesn't exist**